File: Easy/27.PowersOfTwo/solve.rb
def PowersofTwo(num)

	power = 2
	check = 2 ** power 
	while num >= check
		if num == check
			return true
		else
			 
			check = 2 ** (power+=1)
		end
	end
	false
         
end
   
# keep this function call here    
puts PowersofTwo(STDIN.gets)

File: Medium/25.FibonacciChecker/solve.py
import math

def isPerfectSquare(n):
	s = int(math.sqrt(n))
	return s*s == n

def FibonacciChecker(num):
	if isPerfectSquare(5*num*num -4) or isPerfectSquare(5*num*num +4):
		return "yes"
	return "no"
# keep this function call here  
print FibonacciChecker(raw_input())